nixos/modules/system/boot/loader/generic-extlinux-compatible/extlinux-conf-builder.sh

```diff
@@ -95,7 +96,9 @@ addEntry() {
     fi
     echo "  LINUX ../nixos/$(basename $kernel)"
     echo "  INITRD ../nixos/$(basename $initrd)"
-    if [ -d "$dtbDir" ]; then
+    if [ -n "$dtbName" ]; then
+        echo "  FDT ../nixos/$(basename $dtbs)/${dtbName}.dtb"
+    elif [ -d "$dtbDir" ]; then
         echo "  FDTDIR ../nixos/$(basename $dtbs)"
     fi
```

Comment:
The logic here seems a bit screwy. I think the script's `-n` option should fail if `dtbDir` ends up not set.

Reply:
right - but once `-n` is set, we'd still need to pick up from `dtbDir`.

Reply:
pushed a new version, PTAL.